Subject: [PATCH] 2.6.13-rc6-git[67] UML compile fix
From: Peter Zijlstra <a.p.zijlstra@chello.nl>

Hi Jeff, Andrew,

A small patch to make UML build on recent kernels. Still in time for 2.6.13 ?

Signed-off-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
Signed-off-by: Olaf Hering <olh@suse.de>

--- linux-2.6.13-rc6-git7/include/asm-um/page.h~  2005-08-15 21:19:47.000000000 +0200
+++ linux-2.6.13-rc6-git7/include/asm-um/page.h   2005-08-15 23:15:05.000000000 +0200
@@ -104,8 +104,8 @@
  * casting is the right thing, but 32-bit UML can't have 64-bit virtual
  * addresses
  */
-#define __pa(virt) to_phys((void *) (unsigned long) virt)
-#define __va(phys) to_virt((unsigned long) phys)
+#define __pa(virt) to_phys((void *) (unsigned long) (virt))
+#define __va(phys) to_virt((unsigned long) (phys))

 #define page_to_pfn(page) ((page) - mem_map)
 #define pfn_to_page(pfn) (mem_map + (pfn))

-- 
Peter Zijlstra <a.p.zijlstra@chello.nl>

-
To unsubscribe from this list: send the line "unsubscribe linux-kernel" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html
Please read the FAQ at  http://www.tux.org/lkml/

